homeassistant/components/palazzetti/sensor.py

```diff
@@ -0,0 +1,110 @@
+"""Support for Palazzetti sensors."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import UnitOfMass, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from . import PalazzettiConfigEntry
+from .coordinator import PalazzettiDataUpdateCoordinator
+from .entity import PalazzettiEntity
+
+
+@dataclass(frozen=True, kw_only=True)
+class CallableSensorEntityDescription(SensorEntityDescription):
+    """Describes a Palazzetti sensor entity that is read from a `Callable`."""
+
+    value_callable: Callable[[], int | float | str]
+    """The function that returns the state value for this sensor"""
+
+
+@dataclass(frozen=True, kw_only=True)
+class PropertySensorEntityDescription(SensorEntityDescription):
+    """Describes a Palazzetti sensor entity that is read from a `PalazzettiClient` property."""
+
+    presence_flag: None | str
+    """`None` if the sensor is always present, name of a `bool` property of the PalazzettiClient otherwise"""
+
+
+PROPERTY_SENSOR_DESCRIPTIONS: list[PropertySensorEntityDescription] = [
+    PropertySensorEntityDescription(
+        key="pellet_quantity",
+        presence_flag=None,
```

Comment:
this is not really used is it?

Reply:
Not yet. I was planning to add more sensors, but they didn't make it to this PR. I could remove this conditional element and I would have to add it again later.

Reply:
Property Sensors are based have a value that is read from a property on the API client. This property is always set and might have a value of 0 when the device model doesn't have a probe for it. The presence of a probe is made explicit in another bool property, called `presence_flag`. It is used on line 79.

In this PR, the only Property Sensor that I have added is the Pellet Quantity, which is always present. There are other sensors that I will include in an ulterior PR, that need to be added or not depending on a presence_flag.